### PR TITLE
[wip] autogen: update to 2.18.16, fix cross

### LINKED ISSUE #6559: [FTBFS] autogen
### System

* xuname:  
  * Void 4.19.13_1 x86_64 GenuineIntel uptodate FFFF
* package:  
  * autogen 5.18.4

### Expected behavior
Compiles in xbps-src
### Actual behavior
Patch fails to apply. Will not cross compile without patch.
```
=> autogen-5.18.4_8: patching: fix-cross.patch.
1 out of 1 hunk FAILED -- saving rejects to file config/ag_macros.m4.rej
```
How did this happen? Did they change distfiles upstream? There is a checksum though...
### Steps to reproduce the behavior
`xbps-src pkg autogen`

### PR DESCRIPTION
**Pull request by bobertlo**
closes #6559

### CODE REVIEW

File: srcpkgs/autogen/template [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
-hostmakedepends="pkg-config perl"
-makedepends="guile guile-devel libxml2-devel gc-devel"
+configure_args="--disable-static --disable-dependency-tracking --without-libregex"
+hostmakedepends="guile pkg-config perl"
+makedepends="guile guile-devel libatomic_ops-devel libxml2-devel gc-devel"
 depends="guile perl"
-short_desc="The Automated Program Generator"
+short_desc="Automated Program Generator"
 maintainer="Juan RP <xtraeme@voidlinux.org>"
 license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/autogen/template]: GPL-3.0-or-later

File: srcpkgs/autogen/template [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
 homepage="http://autogen.sourceforge.net/"
 distfiles="${GNU_SITE}/${pkgname}/rel${version}/${pkgname}-${version}.tar.gz"
-checksum=3cd9f81a8ae7c6865bb9bbbe16c4307a243a1373d0b315a83571cbba1fff725d
-
+checksum=e23c5bbd0ac83079ae2ef6eb3fd1948fecce718ac853025607a3ab0395538406
 replaces="autogen-docs>=0 autogen-devel>=0"
+_libopts=libopts-42.1.17.tar.gz
+CFLAGS+=" -Wno-error=format-truncation -Wno-error=format-overflow \
```
Comment (maxice8) [inline srcpkgs/autogen/template]: no need for \

**Comment by bobertlo** [comment]
@maxice8 thanks, I changed those.

This should build on travis now, it was cross building for all architectures already on a void system, I think the ubuntu environment was somehow messing with it? All these changes are just making it call host utilities instead of cross, and CFLAGS for newer gcc errors. I still wish I had an arm system to test this on though.

**Comment by bobertlo** [comment]
No, this is definitely an error from the musl bootstrap. cross from glibc works, but building on musl bootrap is broken. The libopts is somehow broken now.

**Comment by Piraty** [comment]
ping

**Comment by bobertlo** [comment]
@Piraty 

I'm starting to think this should just be marked nocross? I really haven't been able to get it to successfully cross compile and @ndowens tried also. I don't know if it would be better to make it nocross or just leave it? If we did make it nocross I would test to make sure it can build natively on arm and push an update.

**Comment by bobertlo** [comment]
Closing because of #16593

**Pull request closed without merging**